providers/default.rb

```diff
@@ -1,16 +1,30 @@
 use_inline_resources
 
 action :update do
+  dir        = new_resource.base_name.split('/')[0..-2].join('/')
+  push_event = new_resource.push_to_datadog || false
+  event_name = new_resource.datadog_event_name
+  event_text = new_resource.datadog_event_text
+  event_key  = new_resource.datadog_event_key
+
   execute 'apt-get-update' do
     command 'apt-get -q update'
     action :nothing
   end
 
-  dir = new_resource.base_name.split('/')[0..-2].join('/')
   directory dir
 
   file new_resource.base_name do
     content new_resource.version
     notifies :run, 'execute[apt-get-update]', :immediately
   end
+
+  if push_event
+    unless event_name.nil? || event_text.nil?
+      require 'datadog/statsd'
+      statsd = Datadog::Statsd.new('localhost', 8125)
+      statsd.event(datadog_event_name, datadog_event_text,
```

Comment:
`bundle exec rake style` seems to tell me 80 characters is the maximum when I have it all on one line